src/main/scala/firrtl/Visitor.scala

```diff
@@ -284,42 +284,31 @@ class Visitor(infoMode: InfoMode) extends FIRRTLBaseVisitor[FirrtlNode] {
     }
   }
 
-  // TODO
-  // - Add mux
-  // - Add validif
   private def visitExp[FirrtlNode](ctx: FIRRTLParser.ExpContext): Expression = {
     val ctx_exp = ctx.exp.asScala
     if (ctx.getChildCount == 1)
       Reference(ctx.getText, UnknownType)
     else
       ctx.getChild(0).getText match {
         case "UInt" =>
-          // This could be better
-          val (width, value) =
-            if (ctx.getChildCount > 4)
-              (IntWidth(string2BigInt(ctx.intLit(0).getText)), string2BigInt(ctx.intLit(1).getText))
-            else {
-              val bigint = string2BigInt(ctx.intLit(0).getText)
-              (IntWidth(BigInt(scala.math.max(bigint.bitLength, 1))), bigint)
-            }
-          UIntLiteral(value, width)
+          if (ctx.getChildCount > 4) {
```

Comment:
I'm OCD today. 4 seems a bit of a magic number, I guess it means here that there are enough fields in the ctx to contain a width. Not worth refactoring.

Reply:
Hahaha yeah, basically "does it have a width". Hopefully this old code will be bypassed soon enough.